com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs

```diff
@@ -767,19 +767,32 @@ static void DrawEmissionContent(SerializedHDLight serialized, Editor owner)
             {
                 serialized.settings.DrawCookie();
 
-                // When directional light use a cookie, it can control the size
-                if (serialized.settings.cookie != null && lightType == HDLightType.Directional)
+                if (serialized.settings.cookie is Texture cookie && cookie != null)
                 {
-                    EditorGUI.indentLevel++;
-                    EditorGUI.BeginChangeCheck();
-                    var size = new Vector2(serialized.shapeWidth.floatValue, serialized.shapeHeight.floatValue);
-                    size = EditorGUILayout.Vector2Field(s_Styles.cookieSize, size);
-                    if (EditorGUI.EndChangeCheck())
+                    // When directional light use a cookie, it can control the size
+                    if (lightType == HDLightType.Directional)
+                    {
+                        EditorGUI.indentLevel++;
+                        EditorGUI.BeginChangeCheck();
+                        var size = new Vector2(serialized.shapeWidth.floatValue, serialized.shapeHeight.floatValue);
+                        size = EditorGUILayout.Vector2Field(s_Styles.cookieSize, size);
+                        if (EditorGUI.EndChangeCheck())
+                        {
+                            serialized.shapeWidth.floatValue = size.x;
+                            serialized.shapeHeight.floatValue = size.y;
+                        }
+                        EditorGUI.indentLevel--;
+                    }
+                    else if (lightType == HDLightType.Point && cookie.dimension != TextureDimension.Cube)
                     {
-                        serialized.shapeWidth.floatValue = size.x;
-                        serialized.shapeHeight.floatValue = size.y;
+                        Debug.LogError($"The cookie texture '{cookie.name}' isn't compatible with the Point Light type. Only Cube textures are supported.");
+                        serialized.settings.cookieProp.objectReferenceValue = null;
```

Comment:
When we clean, it means that user that return to a compatible light will don't have its value anymore. Is it what we want?

Reply:
Can't we just hide the field when it is not relevant ?

Reply:
No, it's the same field for Point and Spot lights and because it's a Texture Field, it can accept any kind of texture dimension.
Btw, the behavior on built-in right now is to set to null the cookie when not compatible (like what I did here).
I asked UX for more information :)

Reply:
In that case, I guess it is ok. Alignment is always better

Reply:
UX approved :) (so with this we're aligned with built-in and URP when their PR land)